kickstarter/android-oss
# WEB-747

# 📲 What

When a username was very long, badges were pushed off screen (see screenshots below)

# 🛠 How

- Setting max number of lines to 1 and putting an ellipsis at the  end of the username
- Fixing the badges to the end of the comment UI

# 👀 See
Before
![Screenshot_20220921_152109](https://user-images.githubusercontent.com/19390326/193049408-c944d3ea-efe5-4740-bcdc-1d4df6a7bb6b.png)

After (short and long usernames)
![LongUsernameExample](https://user-images.githubusercontent.com/19390326/193049402-e303075c-4dc2-4402-b54b-88da3c5bfe3e.png) ![ShortUsernameExample](https://user-images.githubusercontent.com/19390326/193049405-b4927e15-65f2-4ecc-8c6a-6d5dba8c2c05.png) ![youbadge](https://user-images.githubusercontent.com/19390326/193051111-91ed16d1-a5b5-4177-8822-17ef73663f68.png)

# 📋 QA

- Go to comments
- Check that badges are fixed to the right side of the screen
- Check that usernames truncate when they reach a badge or the end of the screen and end with an ellipsis.
# Story 📖

[WEB-747: [Android] UI – User badges broken with long usernames](https://kickstarter.atlassian.net/browse/WEB-747)
